apps/nextjs/src/components/activity/create_activity.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from "react";
+import DatePicker from "react-datepicker";
+
+import { api } from "~/utils/api";
+
+import "react-datepicker/dist/react-datepicker.css";
+
+export default function CreateActivity() {
+  const currentDateTime: Date = new Date();
+  const [selectedDate, setSelectedDate] = useState<Date | null>(
+    currentDateTime,
+  );
+  const [selectedStartDate, setselectedStartDate] = useState<Date | null>(
+    currentDateTime,
+  );
+  const createActivity = api.activity.createActivity.useMutation();
+
+  const [activityData, setActivityData] = useState({
+    name: "",
+    duration: "",
+    leader: "",
+    location: "",
+    selectedDate: currentDateTime,
+    startDate: currentDateTime,
+  });
+
+  const handleInputChange = (field: string, value: string | number) => {
+    setActivityData({
+      ...activityData,
+      [field]: value,
+    });
+  };
+
+  const handleSubmit = () => {
+    if (
+      activityData.name == "" ||
+      activityData.duration == "" ||
+      activityData.leader == "" ||
+      activityData.location == ""
+    ) {
+      alert("You cannot have empty fields!");
+      return;
+    }
+
+    const duration = parseInt(activityData.duration, 10);
+
+    createActivity.mutate({
+      name: activityData.name,
+      day: activityData.selectedDate,
+      durationMinutes: duration,
+      leader: activityData.leader,
+      location: activityData.location,
+      // always creates an object with start date 1970?
+      startTime: activityData.startDate,
+    });
+
+    setActivityData({
+      name: "",
+      duration: "",
+      leader: "",
+      location: "",
+      selectedDate: currentDateTime,
+      startDate: currentDateTime,
+    });
+
+    alert("Submitted!");
+  };
```

Comment:
for the form can you use react-hook-form? you can see an example at https://github.com/uoftblueprint/centre-for-dreams/blob/main/apps/nextjs/src/pages/createpost.tsx

Reply:
fixed